### DISCUSSION TITLE
Faust produce cuts in sound

### DISCUSSION THREAD

**Discussion by SandeepJagdev** [pschatzmann/arduino-audio-tools#1103] [Q&A]
hello, i want to add reverb effect on the sound coming from line input for this i use your guitarx example not it produce cut in sound at output (i also allocate psram). please help. 
here my code is:
#include "AudioTools.h"
#include "AudioLibs/AudioKit.h"
#include "AudioLibs/AudioFaust.h"
#include "guitarix.h"
#include "esp_heap_caps.h"
AudioKitStream kit; 
FaustStream<mydsp> faust(kit); // final output of Faust is kit
StreamCopy copier(faust, kit);  // copy data from kit to faust

// Arduino Setup
void setup(void) {  
  // Open Serial 
  const int limit = 1000000;
heap_caps_malloc_extmem_enable(limit);
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Warning);
if(psramInit()){
        Serial.println("\nThe PSRAM is correctly initialized");
}else{
        Serial.println("\nPSRAM does not work");
}
  // Setup Faust
  auto cfg = faust.defaultConfig();
  faust.begin(cfg);

  // Tube Screemer
  faust.setLabelValue("drive", 0.23);
  faust.setLabelValue("level", -12.6);
  faust.setLabelValue("tone", 489);
  // preamp
  faust.setLabelValue("Pregain", 0.4);
  faust.setLabelValue("Gain", 0.4);
  //jcm 2000
  faust.setLabelValue("Treble", 0.9);
  faust.setLabelValue("Middle", 0.76);
  faust.setLabelValue("Bass", 0.62);
  //Cabinet
  faust.setLabelValue("amount", 100);

  // start I2S
  auto cfg_i2s = kit.defaultConfig(RXTX_MODE);
  cfg_i2s.sample_rate = cfg.sample_rate; 
  cfg_i2s.channels = cfg.channels;
  cfg_i2s.bits_per_sample = cfg.bits_per_sample;
  cfg_i2s.input_device = AUDIO_HAL_ADC_INPUT_LINE2;
  kit.begin(cfg_i2s);

}

// Arduino loop - copy sound to kit 
void loop() {
  copier.copy();
}

**Comment by pschatzmann** [comment]
I don't quite understand your problem: Why don't you use just the [original example](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-faust/streams-i2s-faust_guitarix-i2s/streams-i2s-faust_guitarix-i2s.ino) and adapt it to your requirements....

**Reply by SandeepJagdev** [reply]
original example also produce sound glitches at output. (not working even after PSRAM enabling). please help.

**Reply by pschatzmann** [reply]
What do you mean with sound glitches ?
Maybe you better use the [sound effects provided by the audiotools](https://www.pschatzmann.ch/home/2022/10/14/guitar-effects-with-web-gui/)

**Reply by SandeepJagdev** [reply]
means stream cut and start again 2 times a second (Audio breakup). according to you which is most efficient way to produce reverb. sound effect does not contains reverb. i want to put reverb effect on line input like car DSP do. Please help.

**Comment by pschatzmann** [comment]
You could try the Delay effect. STK has also a couple of implementations and I am pretty sure that Faust has also some...

**Reply by SandeepJagdev** [reply]
fasut has too much audio breakup.please tell how to use stk

**Comment by pschatzmann** [comment]
Did you try the ones from the [documentation](https://faustlibraries.grame.fr/libs/reverbs/) ?
The stk ones can be used like any other effect

**Reply by SandeepJagdev** [reply]
Do you have any arduino example??

**Comment by pschatzmann** [comment]
Yes: the is a link in the [documentaion](https://github.com/pschatzmann/arduino-audio-tools/wiki/Audio-Effects)

**Reply by SandeepJagdev** [reply]
Vs1053 has inbuilt DSP can it make good reverb instead of esp32
